tests/integration/devfile/docker/cmd_docker_devfile_url_test.go

```diff
@@ -241,4 +203,41 @@ var _ = Describe("odo docker devfile url command tests", func() {
 		})
 	})
 
+	Context("Switching pushtarget", func() {
+		It("switch from docker to kube, odo push should display warning", func() {
+			var stdout string
+
+			helper.CmdShouldPass("odo", "create", "nodejs", cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			helper.CmdShouldPass("odo", "url", "create")
+
+			helper.CmdShouldPass("odo", "preference", "set", "pushtarget", "kube", "-f")
+			session := helper.CmdRunner("odo", "push", "--devfile", "devfile.yaml")
```

Comment:
I a curious to know why the output channeling to both ```stdout``` and ```stderr``` instead of one. I feel this is not a good user experience. Is it by design for odo push ?

We have wrapper ```CmdShouldPass``` and ```CmdShouldFail``` around the function ```CmdRunner``` which handles ```stdout``` and ```stderr``` pretty well. So ```push --devfile...``` can be done in the same way for a happy user experience. no ?

Reply:
Since the expected output contains a warning message, and warning message is in the `stderr`. However, this command should pass, but `CmdShouldPass` only handles `stdout`

Reply:
> Since the expected output contains a warning message, and warning message is in the stderr. 

Got it, Warning messages are channeled to ```stderr```. Thanks